src/java/org/apache/cassandra/tcm/Startup.java

```diff
@@ -365,21 +367,29 @@ public static void startup(Supplier<Transformation> initialTransformation, boole
     {
         ClusterMetadata metadata = ClusterMetadata.current();
         NodeId self = metadata.myNodeId();
-        AccordService.startup(self);
 
         // finish in-progress sequences first
         InProgressSequences.finishInProgressSequences(self);
         metadata = ClusterMetadata.current();
 
-        switch (metadata.directory.peerState(self))
+        NodeState startingstate = metadata.directory.peerState(self);
+        switch (startingstate)
+        {
+            case REGISTERED:
+            case LEFT:
+                break;
+            default:
+                AccordService.startup(self);
+        }
+        switch (startingstate)
```

Comment:
I'm a bit confused by this change. Above we're exiting on `REGISTERED` and `LEFT` and then we go back to the `switch` where we do startup on those. It feels like the intention was to only leave the below version in, could you check this out?

Reply:
@ifesdjeen I tried to document this. 

```
// When Accord starts up it needs to check for any historic epochs that it needs to know about (in order
// to handle pending transactions), in order to know what nodes to check with it needs to know what the
// settled placement is (so it knows what peers to reach out to).
```